Database/Constraint/AddTicket Procedure.sql

```diff
@@ -1,31 +1,82 @@
-﻿DROP PROCEDURE if exists sp_add_ticket_type
+﻿/*==============================================================*/
+/* DBMS name:		FestiBase									*/
+/* PDM version:		6											*/
+/* Last edited:		19/12/2017									*/
+/* Author:			Leon Chen									*/
+/* Constraint													*/
+/*==============================================================*/
+
+/*
+Leon Chen
+Procedure to add a new ticket_type to the database
+Constraint:	6
+*/
+
+
+DROP PROCEDURE if exists sp_add_or_update_ticket_type
 GO
-CREATE PROCEDURE sp_add_ticket_type
-@festival_number INT,
-@branch_number INT,
+CREATE PROCEDURE sp_add_or_update_ticket_type
+@festival_company_number INT,
 @ticket_type VARCHAR(50),
 @price money,
 @date_valid_from DATETIME,
-@date_valid_to DATETIME
+@date_valid_to DATETIME,
+@insert bit
 AS
 BEGIN
 	IF @@ROWCOUNT = 0
 		RETURN
 	SET NOCOUNT ON
+	BEGIN TRY
+		EXEC sp_check_ticket_type_start_end_date @festival_company_number, @date_valid_from, @date_valid_to
+
+		IF(@insert = 1)
+		BEGIN
+			INSERT INTO TICKET_TYPE VALUES (@festival_company_number, @ticket_type, @price, @date_valid_from, @date_valid_to)
+		END
+		ELSE BEGIN
+			UPDATE TICKET_TYPE SET festival_company_number = @festival_company_number, ticket_type = @ticket_type, price = @price,
```

Comment:
Wat als de combinatie @festival_company_number, @ticket_type geen resultaten oplevert? Dan komt er 0 rows affected zonder foutmelding.
In de nieuwe template is dit toegevoegd, dat je dan een foutmelding terug geeft.